### ISSUE TITLE
Sampling-free Epistemic Uncertainty Estimation Using Approximated Variance Propagation

### ISSUE THREAD

**Issue by howardyclo** [howardyclo/papernotes#68]
### Metadata
- Authors: Janis Postels, Francesco Ferroni, +2, Federico Tombari
- Organization: Technical University Munich, Autonomous Intelligent Driving GmbH & Google
- Conference: ICCV 2019
- Paper: https://arxiv.org/pdf/1908.00598.pdf
- Code: https://github.com/janisgp/Sampling-free-Epistemic-Uncertainty (Well-written in Keras).

### TL;DR
![](https://i.imgur.com/svDXAZS.png)

Propose an approximation of epistemic uncertainty estimated by MC-dropout (see #67) that does not require sampling (thus reduce computational cost). The whole idea is based on [*error propagation*](https://en.wikipedia.org/wiki/Propagation_of_uncertainty) from physics to estimate the epistemic uncertainty *analytically*.

### Preliminary
- Aleatoric (data) and Epistemic (model) uncertainty
- Why MC-dropout works? Bayesian approximation.
- What is error propagation? [Few examples](https://courses.washington.edu/phys431/propagation_errors_UCh.pdf) & [Full guide].(http://phylabs1.physics.sunysb.edu/introlabs/ReferenceDocs/ErrorAnalysis.pdf)
- Activations after dropout can be well-approximated by Gaussian (section 2.2 in the [Fast dropout training](https://nlp.stanford.edu/pubs/sidaw13fast.pdf) paper).

### TBD.
